test/Tests/Strings/DiscreteCharTest.cs

```diff
@@ -64,6 +64,31 @@ public void BadRanges()
             Assert.True(false);
         }
 
+        [Fact]
+        [Trait("Category", "StringInference")]
+        public void SampleFromUniformCharDistribution()
+        {
+            // 10 chars in distributions
+            const int numChars = 10;
+            const int charExp = 1000;
+            const int numTrials = numChars * charExp; 
+            var dist = DiscreteChar.UniformInRanges("aj");
+
+            var count = new int[numChars];
+            for (var i = 0; i < numTrials; ++i)
+            {
+                count[dist.Sample() - 'a'] += 1;
+            }
+
+            
+            // We could calculate some statistical test (like chi-squared) to verify that distribution is
```

Comment:
I think this is an adequate test in principle; we want to keep tests simple. I would add more samples and have a tighter tolerance. See first test in test/Tests/Core/RandomTest.cs and model it on (part of) that.

Reply:
Modelled test on `RandomTests.RandUniform()`